routes/user.go

```diff
@@ -3775,3 +3776,134 @@ func (fes *APIServer) GetProfileEntryResponseForPublicKeyBytes(publicKeyBytes []
 	}
 	return profileEntryResponse
 }
+
+type GetHoldersForPublicKeyWithLockedBalancesRequest struct {
+	// Either PublicKeyBase58Check or Username can be set by the client to specify
+	// which user we're obtaining posts for
+	// If both are specified, PublicKeyBase58Check will supercede
+	PublicKeyBase58Check string `safeForLogging:"true"`
+	Username             string `safeForLogging:"true"`
+}
+
+type GetHoldersForPublicKeyWithLockedBalancesResponse struct {
+	Holders map[string]*ExtendedBalanceEntryResponse
+}
+
+func (fes *APIServer) GetHoldersForPublicKeyWithLockedBalances(ww http.ResponseWriter, req *http.Request) {
+	decoder := json.NewDecoder(io.LimitReader(req.Body, MaxRequestBodySizeBytes))
+	requestData := GetHoldersForPublicKeyWithLockedBalancesRequest{}
+	if err := decoder.Decode(&requestData); err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf(
+			"GetHoldersForPublicKeyWithLockedBalances: Problem parsing request body: %v", err))
+		return
+	}
+
+	// Get a view
+	utxoView, err := fes.backendServer.GetMempool().GetAugmentedUniversalView()
+	if err != nil {
+		_AddBadRequestError(ww, fmt.Sprintf("GetHoldersForPublicKeyWithLockedBalances: Error getting utxoView: %v", err))
+		return
+	}
+
+	// Decode the public key for which we are fetching hodlers / hodlings.  If public key is not provided, use username
+	var creatorPublicKeyBytes []byte
+	if requestData.PublicKeyBase58Check != "" {
```

Comment:
You may be able to replace this `if else` with `GetPubKeyAndProfileEntryForUsernameOrPublicKeyBase58Check`

Reply:
It replaces the `else` case as well! The other reason I recommend using these is in the event that we want to broadly add more validations. reducing code duplication makes my life easier when we have to introduce new constraints.